cmd/access/node_builder/staked_access_node_builder.go

```diff
@@ -24,14 +27,44 @@ func NewStakedAccessNodeBuilder(anb *FlowAccessNodeBuilder) *StakedAccessNodeBui
 	}
 }
 
+func (fnb *StakedAccessNodeBuilder) InitIDProviders() {
+	fnb.Module("id providers", func(builder cmd.NodeBuilder, node *cmd.NodeConfig) error {
+		idCache, err := p2p.NewProtocolStateIDCache(node.Logger, node.State, fnb.ProtocolEvents)
+		if err != nil {
+			return err
+		}
+
+		fnb.IdentityProvider = idCache
+		// translator
+		fnb.SyncEngineParticipantsProvider = id.NewFilteredIdentifierProvider(
+			filter.And(
+				filter.HasRole(flow.RoleConsensus),
+				filter.Not(filter.HasNodeID(node.Me.NodeID())),
+				p2p.NetworkingSetFilter,
+			),
+			idCache,
+		)
+		fnb.IDTranslator = p2p.NewHierarchicalIDTranslator(idCache, p2p.NewUnstakedNetworkIDTranslator())
+
+		// TODO: NetworkingIdentifierProvider should be the same as the one used in scaffold.go if this AN
+		// doesn't participate in unstaked network.
+		// If it does, then we can just use the default one (peerstoreProvider)
+
+		return nil
+	})
+}
+
 func (builder *StakedAccessNodeBuilder) Initialize() cmd.NodeBuilder {
 
 	ctx, cancel := context.WithCancel(context.Background())
 	builder.Cancel = cancel
 
+	builder.InitIDProviders()
+
 	// for the staked access node, initialize the network used to communicate with the other staked flow nodes
 	// by calling the EnqueueNetworkInit on the base FlowBuilder like any other staked node
-	builder.EnqueueNetworkInit(ctx)
+	// TODO: we should remove this call since we are no longer instantiating two networks
+	// builder.EnqueueNetworkInit(ctx)
```

Comment:
@vishalchangrani

Reply:
yup..will do that. Is there a story for ripping this all out? or should i jsut create one

Reply:
You can proably create one, although it is mentioned in https://github.com/onflow/flow-go/issues/1143